src/controllers/authController.js

```diff
@@ -24,34 +24,63 @@ const path = require('path');
 const constants = require('../utils/constants');
 const adminDao = require('../dao/admin');
 const IdentityProviderDTO = require("../dto/identityProvider");
+const minimatch = require('minimatch');
 
 
 const filePrefix = config.pathToContent;
 
-const fetchAuthJsonContent = async (orgName) => {
+const fetchAuthJsonContent = async (req, orgName) => {
 
+    //use super admin for org creation page login
+    if (req.session.returnTo) {
+        if (minimatch.minimatch(req.session.returnTo, constants.ROUTE.DEVPORTAL_CONFIGURE) ||
+            minimatch.minimatch(req.session.returnTo, constants.ROUTE.DEVPORTAL_ROOT)) {
+            return config.identityProvider;
+        }
+    }
     if (config.mode === constants.DEV_MODE) {
         const authJsonPath = path.join(process.cwd(), filePrefix + '../mock', 'auth.json');
         return JSON.parse(fs.readFileSync(authJsonPath, constants.CHARSET_UTF8));
     }
+    //if no idp per org, use super IDP
+    if (orgName === 'undefined') {
+        return config.identityProvider;
+    }
     try {
         const orgId = await adminDao.getOrgId(orgName);
         const response = await adminDao.getIdentityProvider(orgId);
         if (response.length === 0) {
-            throw new Error(`Failed to fetch identity provider details: ${response.statusText}`);
+            //login from super IDP
+            return config.identityProvider;
         }
         return new IdentityProviderDTO(response[0].dataValues);
     } catch (error) {
         console.error("Failed to fetch identity provider details", error);
-        return {};
+        return config.identityProvider;
     }
 };
 
 const login = async (req, res, next) => {
 
-    const authJsonContent = await fetchAuthJsonContent(req.params.orgName);
-    if (authJsonContent.clientId) {
-        configurePassport(authJsonContent);  // Configure passport dynamically
+    const IDP = await fetchAuthJsonContent(req, req.params.orgName);
+    if (IDP.clientId) {
+        //fetch claim names from DB
+        const orgName = req.params.orgName;
```

Comment:
Assign orgName before the method so it can be used in fetchAuthJsonContent as well.